parser/src/parser/context.rs

```diff
@@ -263,17 +285,16 @@ impl ParserContext {
     pub fn eat_int(&mut self) -> Option<(PositiveNumber, Span)> {
         if let Some(SpannedToken {
             token: Token::Int(_), ..
-        }) = self.tokens.last()
+        }) = self.curr()
         {
-            let token = self.tokens.pop().unwrap();
             if let SpannedToken {
                 token: Token::Int(value),
                 span,
-            } = token
+            } = self.bump().unwrap()
             {
                 return Some((PositiveNumber { value }, span));
             } else {
-                unimplemented!()
+                unreachable!("Impossible to reach area, unless a change broke the parser.")
```

Comment:
This is more of a comment than an error message.
I know we are never going to reach this case but can we update this message so that it would be helpful in the alternate reality where we hit this error?
```suggestion
                unreachable!("The parser encountered a non-spanned token. Please open an issue on github if you see this message")
```